azure/functions/decorators/function_app.py

```diff
@@ -302,22 +264,28 @@ def decorator():
 
         return wrap
 
-    def _add_http_app(self,
-                      http_middleware: Union[
-                          AsgiMiddleware, WsgiMiddleware]) -> None:
-        """Add a Wsgi or Asgi app integrated http function.
 
-        :param http_middleware: :class:`AsgiMiddleware` or
-        :class:`WsgiMiddleware` instance.
+class HttpFunctionsAuthLevelMixin(ABC):
+    """Interface to extend for enabling function app level http
+    authorization level setting"""
 
-        :return: None
+    def __init__(self, auth_level: Union[AuthLevel, str], *args, **kwargs):
+        self._auth_level = AuthLevel[auth_level] \
```

Comment:
What if auth_level is None?

Reply:
then authlevel will not be included in binding.json and host will set default auth level to Function